Habitica/src/main/java/com/habitrpg/android/habitica/ui/viewmodels/NotificationsViewModel.kt

```diff
@@ -0,0 +1,144 @@
+                .distinctUntilChanged()
+    }
+
+    fun allNotificationsSeen(): Flowable<Boolean> {
+        return getNotifications()
+                .map { it.all { notification -> notification.seen == true } }
+                .distinctUntilChanged()
+    }
+
+    fun refreshNotifications(): Flowable<*> {
+        return userRepository.retrieveUser(withTasks = false, forced = true)
+    }
+
+    private fun filterSupportedTypes(notifications: List<Notification>): List<Notification> {
+        return notifications.filter { supportedNotificationTypes.contains(it.type) }
+    }
+
+    fun isPartyMessage(data: NewChatMessageData?): Boolean {
+        if (party == null || data?.group?.id == null) {
+            return false
+        }
+
+        return party?.id == data.group?.id
+    }
+
+    fun dismissNotification(notification: Notification) {
+        disposable.add(userRepository.readNotification(notification.id)
+                .subscribe(Consumer {}, RxErrorHandler.handleEmptyError()))
+    }
+
+    fun dismissAllNotifications(notifications: List<Notification>) {
+        if (notifications.isEmpty()) {
+            return
+        }
+
+        val notificationIds = HashMap<String, List<String>>()
+        notificationIds["notificationIds"] = notifications.map { notification -> notification.id }
+
+        disposable.add(userRepository.readNotifications(notificationIds)
+                .subscribe(Consumer {}, RxErrorHandler.handleEmptyError()))
+    }
+
+    fun markNotificationsAsSeen(notifications: List<Notification>) {
+        val unseenIds = notifications.filter { notification -> notification.seen != true }
+                .map { notification -> notification.id }
+
+        if (unseenIds.isEmpty()) {
+            return
+        }
+
+        val notificationIds = HashMap<String, List<String>>()
+        notificationIds["notificationIds"] = unseenIds
+
+        disposable.add(userRepository.seeNotifications(notificationIds)
+                .subscribe(Consumer {}, RxErrorHandler.handleEmptyError()))
+    }
+
+    fun click(notificationId: String, navController: MainNavigationController) {
+        val notification = notificationsManager.getNotification(notificationId) ?: return
+
+        dismissNotification(notification)
+
+        when (notification.type) {
+            Notification.Type.NEW_STUFF.type -> navController.navigate(R.id.newsFragment)
+            Notification.Type.NEW_CHAT_MESSAGE.type -> clickNewChatMessage(notification, navController)
+            Notification.Type.NEW_MYSTERY_ITEMS.type -> navController.navigate(R.id.itemsFragment)
+            Notification.Type.UNALLOCATED_STATS_POINTS.type -> navController.navigate(R.id.statsFragment)
+        }
+    }
+
+    private fun clickNewChatMessage(notification: Notification, navController: MainNavigationController) {
+        val data = notification.data as? NewChatMessageData
+        if (isPartyMessage(data)) {
+            navController.navigate(R.id.partyFragment)
```

Comment:
I guess there's currently no way to navigate straight to the chat view of party/guild fragment? And same with items fragment special view?

Reply:
Maybe could add an argument to open a specific view in the fragment? Or is there some other way to achieve it?